sharkio-dev/sharkio
# fix: issue #223 - rename the directories client and server

## Description

Renamed 'traffic-dashboard' and 'traffic-sniffer' directories to 'client' and 'server' respectively.

## Related Issues or Pull Requests

issue #223 

## Proposed Changes

Along with renaming directories, I made changes to the following files. 
fyi: client means 'traffic-dashboard', server means 'traffic-sniffer'

-  **dashboard-main.js**  --- rename --- import {...} from  `client`
- **sniffer-main.js** --- rename --- import {...} from `server`
- **docker-compose.yml** --- rename --- `backend/build: ./server ` and `frontend/build : ./client`
- **package-lock.json** --- rename --- 'packages : {dependencies : {`"client": "file: client", "server": "file: server"}` }'
- **package.json** --- rename --- 

"scripts" : {
```
  "dashboard": "cd ./client && npm run dev",
    "sniffer": "cd ./server && npm run dev",
    "build:dashboard": "cd ./client && npm run build",
    "build:sniffer": "cd ./server && npm run build",
    "format:dashboard": "cd ./client && npm run format",
    "format:sniffer": "cd ./server && npm run format",
    "build": "concurrently --kill-others \"cd ./server && npm run build\"  \"cd ./client && npm run build\"",
    "dev": "concurrently --kill-others \"cd ./server && npm run dev\"  \"cd ./client && npm run dev\"",
    "dev2": "concurrently --kill-others \"cd ./server && npm run dev\"  \"cd ./client && npm run dev2\"",
    "demo": "concurrently --kill-others \"nodemon dummy-server.js\" \"npm run dev\"",
    "format": "prettier -w . && git add .",
    "test:server": "cd ./server && npm test",
    "test:client": "cd ./client && npm test",
    "test": "concurrently \"npm run test:server\" \"npm run test:client\"",
```
}, 

"dependencies" : {
```
    "client": "file:client",
    "server": "file:server",
```
}
  "files": [
```
    "server",
    "client"
```
  ],
  "bundleDependencies": [
```
    "client",
    "server",
```
  ],

- **.gitignore** --- rename --- 
```
server/.env
server/node_modules/*
client/node_modules/*
server/node_modules/.bin/mime
```
- **readme.md** --- rename --- under _running in development_
For the backend
`- cd into server.`
For the frontend
`- cd into client.`


## Screenshots (if applicable)

If the changes are visual in nature, please provide screenshots or GIFs that showcase the before and after effects.

## How to Test

Outline the steps to test the changes made in this pull request. Provide as much detail as necessary so that reviewers can easily verify the changes.
- `npm run test`

## Additional Notes

Include any other relevant information, context, or considerations related to this pull request.

## Checklist

Please make sure you've done the following before submitting this pull request:

- [ ✅ ] Ran all existing tests to ensure they pass.
- [ ✅ ] Added new tests to cover the changes (if applicable).
- [ ✅ ] Reviewed the code for any potential issues or bugs.
- [ ✅ ] Updated the documentation (if needed) to reflect the changes.
- [ ✅ ] Squashed your commits into a single logical commit (if necessary).
